docs/plugins/development/creating-text-analysis-plugins.asciidoc

```diff
@@ -0,0 +1,129 @@
+[[creating-text-analysis-plugins]]
+=== Creating text analysis plugins
+
+Text analysis plugins provide {es} with custom Lucene analyzers, token filters, 
+character filters, and tokenizers. 
+
+[discrete]
+==== Plugin file structure
+
+Plugins are ZIP files composed of JAR files and two metadata files:
+
+* `stable-plugin-descriptor.properties` - a Java properties file that describes 
+the plugin. Refer to <<stable-plugin-descriptor.properties>>.
+* `named_components.json` - a JSON file mapping interfaces to key-value pairs 
+of component names and implementation classes.
+
+Note that only JAR files at the root of the plugin are added to the classpath
+for the plugin. If you need other resources, package them into a resources JAR.
+
+[discrete]
+==== The analysis plugin API
+
+Text analysis plugins can be written against the analysis plugin API. This API
+includes:
+
+* the entirety of the core Lucene module
+* the {es} `plugin-api`, which includes important annotations for the plugin 
+framework
+* `plugin-analysis-api`, which contains the classes that an analysis plugin can 
+implement.
+
+The advantage of using this API is that the resulting plugin is
+forward-compatible with new minor versions of {es}. While elements may be added
+to the API, existing element will not be changed or removed, except in a new
+major release. In other words, once you have a working artifact, you can re-use
+it when you upgrade {es} to a new bugfix or minor version.
+
+It's also possible to create text analysis plugins using the method described in
+<<creating-non-text-analysis-plugins>>. Howeverm plugins created using that 
+method are pinned to specific version of {es}. You will need to recompile them
+when upgrading{es}.
```

Comment:
I would move this to the end of the paragraph. so that does a reader is focused on the new API here

Reply:
Done.